src/systems/thermal/Thermal.cc

```diff
@@ -48,14 +51,60 @@ void Thermal::Configure(const Entity &_entity,
     gazebo::EntityComponentManager &_ecm,
     gazebo::EventManager & /*_eventMgr*/)
 {
-  if (!_sdf->HasElement("temperature"))
+  const std::string temperatureTag = "temperature";
+  const std::string heatSignatureTag = "heat_signature";
+  const std::string minTempTag = "min_temp";
+  const std::string maxTempTag = "max_temp";
+
+  if (_sdf->HasElement(temperatureTag) && _sdf->HasElement(heatSignatureTag))
+  {
+    ignwarn << "Both <" << temperatureTag << "> and <" << heatSignatureTag
+           << "> were specified, but the thermal system only uses one. "
+           << "<" << heatSignatureTag << "> will be used.\n";
+  }
+
+  if (_sdf->HasElement(heatSignatureTag))
+  {
+    std::string heatSignature = _sdf->Get<std::string>(heatSignatureTag);
+
+    // check if this is a fuel model (if so, use the heat signature path in
+    // fuel's local cache instead of the user-specified path)
+    ignition::fuel_tools::FuelClient fuelClient;
+    std::string fuelPath;
+    if (fuelClient.CachedModelFile(ignition::common::URI(heatSignature),
+          fuelPath))
+      _ecm.CreateComponent(_entity, components::SourceFilePath(fuelPath));
```

Comment:
is this for converting URIs to local paths? You can use `asFullPath` as done here: https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo4/src/rendering/SceneManager.cc#L401

Reply:
I used `asFullPath` in https://github.com/ignitionrobotics/ign-gazebo/pull/498/commits/14f327bda9a58c47b0ff82a782b463fbf6c82ea1. However, with this change, the plugin integration tests need to be updated. Now, if an invalid path to a file is given, the plugin won't be loaded for this entity: https://github.com/ignitionrobotics/ign-gazebo/blob/14f327bda9a58c47b0ff82a782b463fbf6c82ea1/src/systems/thermal/Thermal.cc#L73-L80

I think the issue is that if a user enters an absolute path to a texture on their machine, like `/path/to/texture.png` (unix-style path), then this path won't work on windows because windows uses the `\` delimiter instead of `/`. So, I am trying to figure out a way to write a test SDF file that is platform-independent, so that I can update the tests to work for valid absolute paths.

Reply:
Tests should be working now after https://github.com/ignitionrobotics/ign-gazebo/pull/498/commits/d2be46cfb965ac0fe7f0316a4c90a2720adc5c49. I'm waiting for https://github.com/ignitionrobotics/ign-gazebo/pull/600 to be merged before marking this as resolved.

Reply:
#600 was merged, so I think this is resolved.